src/Adapter/Profile/QueryHandler/GetProfileForEditingHandler.php

```diff
@@ -26,28 +26,56 @@
 
 namespace PrestaShop\PrestaShop\Adapter\Profile\QueryHandler;
 
+use PrestaShop\PrestaShop\Adapter\Domain\AbstractObjectModelHandler;
 use PrestaShop\PrestaShop\Core\Domain\Profile\Exception\ProfileNotFoundException;
 use PrestaShop\PrestaShop\Core\Domain\Profile\Query\GetProfileForEditing;
 use PrestaShop\PrestaShop\Core\Domain\Profile\QueryHandler\GetProfileForEditingHandlerInterface;
 use PrestaShop\PrestaShop\Core\Domain\Profile\QueryResult\EditableProfile;
 use PrestaShop\PrestaShop\Core\Domain\Profile\ValueObject\ProfileId;
+use PrestaShop\PrestaShop\Core\Image\Parser\ImageTagSourceParser;
+use PrestaShop\PrestaShop\Core\Image\Parser\ImageTagSourceParserInterface;
 use Profile;
 
 /**
  * Gets Profile for editing using legacy object model
  */
-final class GetProfileForEditingHandler implements GetProfileForEditingHandlerInterface
+final class GetProfileForEditingHandler extends AbstractObjectModelHandler implements GetProfileForEditingHandlerInterface
 {
+    /**
+     * @var ImageTagSourceParserInterface
+     */
+    private $imageTagSourceParser;
+    /**
+     * @var string
+     */
+    private $imgDir;
+
+    /**
+     * @param ImageTagSourceParserInterface|null $imageTagSourceParser
+     * @param string $imgDir
+     */
+    public function __construct(
+        ImageTagSourceParserInterface $imageTagSourceParser = null,
+        string $imgDir = _PS_PROFILE_IMG_DIR_
+    ) {
+        $this->imageTagSourceParser = $imageTagSourceParser ?? new ImageTagSourceParser(__PS_BASE_URI__);
```

Comment:
Maybe we should add a deprecate notice that `null` won't be allowed in next major?

Reply:
@jolelievre Have you got an example ?

Reply:
For once I do ^^
https://github.com/PrestaShop/PrestaShop/blob/42b365326376591dd471359497b5d41fe1942279/src/Core/Grid/GridFactory.php#L78

Reply:
I admit we don't do it everytime we should, but it's a good practice to help us identify the pieces we will have to remove in the next major

Reply:
Done :heavy_check_mark: